addons/mail/static/src/js/mail.js

```diff
@@ -1167,8 +1191,26 @@ var ComposeMessage = Attachment.extend ({
 
         return suggested_partners;
     },
-
+    /**
+     * preprocess_message_post: Used for @ mention functionality
+     * If mention partner remove from textarea we don't send mail to those partner.
+     * Make <span> for mentioned parners that is convert into link at read time.
+     */
+    preprocess_message_post: function() {
```

Comment:
Idem, extract this method and put it in `mail_mention.js`. It should generic to be reused in the timeline view and the next inbox (slack view).